contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -672,12 +741,26 @@ bool USDScene::hasAttribute( const SceneInterface::Name &name ) const
 		pxr::TfToken kind;
 		return model.GetKind( &kind );
 	}
-	else if( pxr::UsdAttribute attribute = m_location->prim.GetAttribute( AttributeAlgo::toUSD( name.string() ) ) )
+	else if( AttributeAlgo::findUSDAttribute( m_location->prim, name.string() ) )
 	{
-		return isCortexAttribute( attribute );
+		return true;
 	}
 	else
 	{
+		std::vector< pxr::UsdPrim > primList = { m_location->prim };
+		pxr::UsdShadeMaterial mat = pxr::UsdShadeMaterialBindingAPI::ComputeBoundMaterials(	primList )[0];
```

Comment:
Why do we use `ComputeBoundMaterials()` rather than `ComputeBoundMaterial()` (singular)? The docs suggest the plural form is optimised for multiple queries, but we're only making a single one. The [singular form](https://graphics.pixar.com/usd/release/api/class_usd_shade_material_binding_a_p_i.html#a5602550a78f68f6a1d1bf62187e22edd) allows us to pass a `BindingsCache` and a `CollectionQueryCache` that sound like they might be quite important for maintaining performance for large scenes, so perhaps we should be doing that instead? It seems like we could store those caches quite conveniently on `USDScene::IO` so they can be shared between all locations? This comments applies to various spots in the other attribute functions as well.

Reply:
Hmm, I must have been looking at the static version for some reason.  The singular form does seem better.  Addressed in 938fbac